pkg/kube/krt/collection_test.go

```diff
@@ -237,6 +237,31 @@ func TestCollectionInitialState(t *testing.T) {
 	assert.Equal(t, fetcherSorted(SimpleEndpoints)(), []SimpleEndpoint{{"pod", "svc", "namespace", "1.2.3.4"}})
 }
 
+func TestCollectionHandlerSync(t *testing.T) {
```

Comment:
A negative test seems equally or more important to prevent a "marked ready early" bug. I'm not sure how to formulate one though besides having two routines race between `WaitUntilSync(); signal` and Register(() { signal })` and ensure the ordering is correct. That probably wouldn't even fail consistently when things are broken.

Reply:
This current test does/can detect 'marked ready early'. If it is marked early, that means we did it before we sent the event through which means `gotEvent.Load()` will be false. It is racy though, detects about 12% of time on my machine